ee/fleetd-chrome/src/tables/privacy_preferences.ts

```diff
@@ -0,0 +1,100 @@
+import Table from "./Table";
+
+export default class TablePrivacyPreferences extends Table {
+  // expose properties available from the chrome.privacy API as a virtual osquery "table"
+  // https://developer.chrome.com/docs/extensions/reference/privacy/
+
+  name = "privacy_preferences";
+  propertyAPIs = {
+    // all of type `types.ChromeSetting<boolean>` with default `true` unless otherwise specified
+
+    // though all of these properties are documented, some have not been added to the
+    // typings we are using: https://www.npmjs.com/package/@types/chrome and must be `@ts-ignore`d
+
+    // network
+    network_prediction_enabled: chrome.privacy.network.networkPredictionEnabled,
+    // type `types.ChromeSetting<IPHandlingPolicy>`, default "default"
+    // IPHandlingPolicy:
+    // "default" | "default_public_and_private_interfaces" |
+    // "default_public_interface_only" | "disable_non_proxied_udp"
+    web_rtc_ip_handling_policy: chrome.privacy.network.webRTCIPHandlingPolicy,
+
+    // services
+    autofill_address_enabled: chrome.privacy.services.autofillAddressEnabled,
+    autofill_credit_card_enabled:
+      chrome.privacy.services.autofillCreditCardEnabled,
+    // DEPRECATED and replaced with above two properties
+    autofill_enabled: chrome.privacy.services.autofillEnabled,
+    save_passwords_enabled: chrome.privacy.services.passwordSavingEnabled,
+    safe_browsing_enabled: chrome.privacy.services.safeBrowsingEnabled,
+    // default false
+    safe_browsing_extended_reporting_enabled:
+      chrome.privacy.services.safeBrowsingExtendedReportingEnabled,
+    search_suggest_enabled: chrome.privacy.services.searchSuggestEnabled,
+    // default false
+    spelling_service_enabled: chrome.privacy.services.spellingServiceEnabled,
+    translation_service_enabled:
+      chrome.privacy.services.translationServiceEnabled,
+
+    // websites
+    // @ts-ignore
+    ad_measurement_enabled: chrome.privacy.websites.adMeasurementEnabled,
+    // default false
+    do_not_track_enabled: chrome.privacy.websites.doNotTrackEnabled,
+    // @ts-ignore
+    fledge_enabled: chrome.privacy.websites.fledgeEnabled,
+    hyperlink_auditing_enabled:
+      chrome.privacy.websites.hyperlinkAuditingEnabled,
+    // @ts-ignore, DEPRECATED
+    privacy_sandbox_enabled: chrome.privacy.websites.privacySandboxEnabled,
+    // WINDOWS AND CHROMEOS ONLY
```

Comment:
What does Windows mean in this context?

Reply:
I take it to mean Chrome browser or ChromeOS Flex on a Windows machine (maybe for some reason a customer would want to query a simulated Chromebook hosted on a PC?). The Windows capability isn't needed for development, no, I just figured start with supporting as much as possible, and we can restrict however we see fit – @zhumo?

Reply:
> I take it to mean Chrome browser or ChromeOS Flex on a Windows machine

My guess is that a device running ChromeOS Flex will report itself as "Chrome OS".
And "Windows" will only be returned as platform when running the extension in a Chrome browser on a Windows OS.

Reply:
Hey all, we don't recommend using the chrome extension on Chrome browers (https://fleetdm.com/docs/using-fleet/chromeos#adding-chromeos-hosts-to-fleet). Only on ChromeOS computers. So I think we don't need to consider/support that case.

Reply:
@zhumo got it, thanks